x-pack/plugin/autoscaling/src/internalClusterTest/java/org/elasticsearch/xpack/autoscaling/AutoscalingCountTestDeciderService.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.autoscaling;
+
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.cluster.node.DiscoveryNodeRole;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xpack.autoscaling.capacity.AutoscalingDeciderContext;
+import org.elasticsearch.xpack.autoscaling.capacity.AutoscalingDeciderResult;
+import org.elasticsearch.xpack.autoscaling.capacity.AutoscalingDeciderService;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class AutoscalingCountTestDeciderService implements AutoscalingDeciderService {
+
+    public static final String NAME = "count";
+
+    private final AtomicInteger counter = new AtomicInteger();
+
+    @Override
+    public String name() {
+        return NAME;
+    }
+
+    @Override
+    public AutoscalingDeciderResult scale(Settings configuration, AutoscalingDeciderContext context) {
+        return new AutoscalingDeciderResult(null, new CountReason(counter.incrementAndGet()));
+    }
+
+    @Override
+    public List<Setting<?>> deciderSettings() {
+        return org.elasticsearch.core.List.of();
+    }
+
+    @Override
+    public List<DiscoveryNodeRole> roles() {
+        return Collections.unmodifiableList(new ArrayList<>(DiscoveryNode.getPossibleRoles()));
```

Comment:
NIT: Looks like `org.elasticsearch.core.List.of(...)` produces unmodifiable list as well so above could be simplified.

Reply:
I am not sure I follow how this could be used here. `List.of` expects individual elements, but here we have a collection of roles that need conversion to an unmodifiable list.